src/common/tracer.h

```diff
@@ -45,8 +46,41 @@ class Tracer {
 
 };
 
-void encode(const jspan_context& span, ceph::buffer::list& bl, uint64_t f = 0);
-void decode(jspan_context& span_ctx, ceph::buffer::list::const_iterator& bl);
+inline void encode(const jspan_context& span_ctx, bufferlist& bl, uint64_t f = 0) {
+  ENCODE_START(1, 1, bl);
+  using namespace opentelemetry;
+  using namespace trace;
+  auto is_valid = span_ctx.IsValid();
```

Comment:
This boolean needs a bunch of extra bytes (version, compat, size) coming from `ENCODE_START`. The idea might be to make `is_valid()` a part of the `MOSDOp`.

Not saying a word whether optimizing this is worth the effort.

Reply:
i will make this change in a separate commit. so we can decide whether it is worthwhile the effort

Reply:
done in 4164264f1d76d5fbc5f81ff06cced38ff03487fc